st2common/st2common/models/api/keyvalue.py

```diff
@@ -44,6 +53,16 @@ class KeyValuePairAPI(BaseAPI):
                 'type': 'string',
                 'required': True
             },
+            'secret': {
```

Comment:
You have 2 properties secret and encrypted so I have a few questions -

1. If a key is a secret it is always encrypted?
2. If a key is encrypted does it have to be a secret?
3. Do we attribute any special semantic meaning to secret? i.e. because a key is a secret it will be handled differently?